### PR TITLE
Work around when master is exactly on a tag name.

### PR DESCRIPTION
**Pull request by kacf**
If we have requested to build master in the build parameters, we want
a SHA-based package version, even if it matches the tag exactly. This
is also expected by our tests, which fail without this fix.

However, the reported version by the tool should be the tag name, if
it's possible. So testing for this when building master is wrong,
because it can be both a tag name and a SHA, depending on what is
built.

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

**Review by oleorhagen — APPROVED** [review]

**Comment by lluiscampos** [comment]
This fixes [QA-276](https://tracker.mender.io/browse/QA-263), right?

**Review by lluiscampos — APPROVED** [review]

**Comment by kacf** [comment]
Correct, I will include the ticket reference before I merge.

**Comment by kacf** [comment]
@lluiscampos: I had to change even more stuff here, does it make sense to remove that part?

**Comment by kacf** [comment]
This is blocking release progress, and it's only a test which has changed since the last review, so I will merge this, but please take a look if you agree afterwards.

**Comment by kacf** [comment]
> Correct, I will include the ticket reference before I merge.

Bah, I forgot this. But the ticket is updated.

**Pull request merged**
